### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by AlselwiAisha**
### Description:

- Use the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as my backend.
- Use [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make my HTTP requests to the backend
- Fetch my books from the Bookstore API after the component has rendered
- Refactor my add book and remove book features to persist my changes in the server.

### CODE REVIEW

File: src/redux/books/booksSlise.js [hunk 1]
```diff
@@ -1,37 +1,89 @@
...
+        state.error = action.payload;
+      })
+      .addCase(removeBook.pending, (state) => {
+        state.isLoading = true;
+      })
+      .addCase(removeBook.fulfilled, (state, action) => {
+        state.isLoading = false;
+        const itemId = action.payload.item_id;
+        state.books = state.books.filter((book) => book.item_id !== itemId);
+      })
```
Comment (Johnadibe) [inline src/redux/books/booksSlise.js]: > - [x] Well done, @AlselwiAisha! I've observed that when I remove a book from both the list and the API, and then refresh the browser, the state doesn't seem to reflect the change. Could you please address this? Resolving this matter will significantly enhance the user experience. 👍
> 
> [screen-capture.webm](https://github.com/AlselwiAisha/Bookstore/assets/40009354/0c869008-60d1-4535-ac73-b2440904f991)
>

File: src/redux/books/booksSlise.js [hunk 2]
```diff
@@ -1,37 +1,89 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+/* eslint no-param-reassign: ["error", { "props": false }] */
```
Comment (Johnadibe) [inline src/redux/books/booksSlise.js]: > - [x] Hello @AlselwiAisha, I'd like to emphasize the importance of maintaining our linter configurations intact. Linters serve as valuable tools in our development workflow; they help us maintain consistent code quality, identify potential issues, and ensure code adherence to best practices. Disabling these linters should not be our approach. Instead, I kindly request that we address the issue at hand in a more constructive manner. By doing so, we can maintain the integrity of our coding standards while resolving any specific concerns effectively. This approach aligns with our commitment to producing high-quality code.👍🚀

File: src/components/AddNewBook.js [hunk 3]
```diff
@@ -18,10 +18,14 @@ const AddNew = () => {
...
-    setAuthor('');
-    setCategory('');
-    setTitle('');
+    if (title && author && category) {
+      dispatch(addBook(data));
+      setAuthor('');
+      setCategory('');
+      setTitle('');
+    } else {
+      alert('Fill all fields, please!');
```
Comment (Johnadibe) [inline src/components/AddNewBook.js]: > - [x] I noticed that you've added the` alert()` method in your implementation. While it's a useful tool for debugging and quick testing, it's important to consider its impact on the overall user experience and code maintainability, especially in production-level code. To enhance your code, I recommend removing the `alert()` method and considering an alternative approach for handling user interactions.👍

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️
### Hi @AlselwiAisha👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Highlights
✅ You used GitFlow correctly 👍🏽
✅ Linter checks passed✔️
✅ Used Axios with createAsyncThunk to make your HTTP requests to the backend✔
✅ Fetched books from the Bookstore API after the component has rendered✔
✅ Refactored add book features to persist changes in the server✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @AlselwiAisha , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Requested changes implemented 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**